model/checkin.json

```diff
@@ -0,0 +1,136 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Checkin",
+  "description": "An Elastic Agent checkin to Fleet",
+  "type": "object",
+  "properties": {
+    "_id": {
+      "description": "The unique identifier for the Elastic Agent checkin",
+      "type": "string",
+      "format": "uuid"
+    },
+    "@timestamp": {
+      "description": "Date/time the checkin was created",
+      "type": "string",
+      "format": "date-time"
+    },
+    "agent": {
+      "title": "Checkin Agent",
+      "description": "The current metadata information for the Elastic Agent",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The ID for the Elastic Agent",
+          "type": "string",
+          "format": "uuid"
+        },
+        "version": {
+          "description": "The version of the Elastic Agent",
+          "type": "string"
+        }
+      },
+      "required": [
+        "id",
+        "version"
+      ]
+    },
+    "host": {
+      "title": "Checkin Host",
+      "description": "The current host information for the Elastic Agent",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The ID of the host",
+          "type": "string"
+        },
+        "architecture": {
+          "description": "The architecture for the Elastic Agent",
+          "type": "string"
+        },
+        "name": {
+          "description": "The hostname of the Elastic Agent",
+          "type": "string"
+        }
+      },
+      "required": [
+        "id",
+        "architecture",
+        "name"
+      ]
+    },
+    "status": {
+      "description": "The current overall status of the Elastic Agent",
+      "type": "string",
+      "enum": ["enrolling", "healthy", "error", "degraded", "offline", "unenrolling", "upgrading"]
+    },
+    "message": {
+      "description": "The current overall status message of the Elastic Agent",
+      "type": "string"
+    },
+    "policy": {
+      "title": "Checkin Policy",
+      "description": "The current status of the applied policy",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The ID for the policy",
+          "type": "string",
+          "format": "uuid"
+        },
+        "revision": {
+          "description": "The revision of the policy",
+          "type": "integer"
+        },
+        "inputs": {
+          "title": "Checkin Policy Input",
+          "description": "The current input status per policy",
+          "type": "array",
```

Comment:
One thing we should be careful here is if we index this or not. If we do, each array entry becomes its own document in Lucene. So if we only need the blob for retrieval that would be best. It still applies that we should have it in the schema as it is here.

Reply:
At the moment, I don't know if we can keep it as a blob. When we do work on the coordinator piece this might need to be indexed.

Or with a transform that updates the latest status, then only the result of that would need to be indexed.

Reply:
We probably want to be able to search for a particular input status, but maybe not in the logs, more in a document that will be created with a transform